sources/packages/api/src/route-controllers/education-program/education-program.controller.ts

```diff
@@ -369,4 +410,64 @@ export class EducationProgramController {
       },
     );
   }
+
+  /**
+   * Ministry user approve's a pending program.
+   * @param programId program id.
+   * @param institutionId institution id.
+   * @UserToken userToken
+   * @Body payload
+   */
+  @AllowAuthorizedParty(AuthorizedParties.aest)
+  @Groups(UserGroups.AESTUser)
+  @Patch(":programId/institution/:institutionId/approve/aest")
+  async approveProgram(
+    @UserToken() userToken: IUserToken,
+    @Param("programId") programId: number,
+    @Param("institutionId") institutionId: number,
+    @Body() payload: ApproveProgram,
+  ): Promise<void> {
+    const updatedProgram = await this.programService.approveEducationProgram(
+      programId,
+      userToken.userId,
+      payload,
+    );
+    /**mapping the note added for approved program to institution notes**/
+    if (updatedProgram.programNote) {
+      await this.institutionService.saveInstitutionNote(
```

Comment:
We need refactor the below method. If I am not wrong it will select the entire institution object from DB alongside will all the related notes and its properties. Was it not supposed to be a simple insert in the table?

![image](https://user-images.githubusercontent.com/61259237/153680328-6b859fe9-192b-41bc-870e-8c842546072a.png)